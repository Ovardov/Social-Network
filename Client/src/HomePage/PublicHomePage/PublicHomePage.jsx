import React from 'react';
import { Link } from 'react-router-dom';
import styles from './public-home-page.module.scss';

function PublicHomePage() {
    return (
        <section className={styles.container}>
            <section className={styles.content}>
                <img src="home-page.svg" alt="Friends" />

                <section className={styles.info}>
                    <h1 className={styles.name}>Social Network</h1>

                    <section className={styles.buttons}>
                        <Link to='/register' className="button">Register</Link>
                        <Link to='/login' className="button">Login</Link>
                    </section>
                </section>
            </section>
        </section>
    )
}

export default PublicHomePage;

