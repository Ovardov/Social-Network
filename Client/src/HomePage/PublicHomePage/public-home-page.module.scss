.container {
    min-height: 100vh;
    width: 100%;
    background: -webkit-linear-gradient(40deg, #FFFFFF 50%, var(--color-blue-main));
    display: flex;
    justify-content: center;
    align-items: center;

    & .content {
        width: 100%;
        height: 100vh;
        display: flex;
        justify-content: space-evenly;
        align-items: center;

            & img {
                height: 30em;
                width: 30em;
            }

        & .info {
            text-align: center;

            & .name {
                font-size: 5rem;
                font-weight: bold;
                color: #fcfcfc;
                line-height: 1;
                text-shadow: 3px 5px var(--color-blue-main);
            }

            & .buttons {
                display: flex;
                justify-content: center;
                align-items: center;

                & a {
                    display: block;
                    padding: 0.6em 1.6em;
                    text-transform: uppercase;
                    width: 10em;

                    &:first-of-type {
                        margin-right: var(--spacing-layout-elements);
                    }
                }
            }
        }
    }
}