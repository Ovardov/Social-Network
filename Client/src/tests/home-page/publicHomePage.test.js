import React from 'react';
import renderer from 'react-test-renderer';
import PublicHomePage from '../../components/HomePage/PublicHomePage/PublicHomePage';

jest.mock("react-router-dom", () => ({
    Link: "Link"
}));

describe('PublicHomePage component', () => {
    test('should render correctly if user is not logged', () => {

        const component = renderer.create(
            <PublicHomePage />
        );

        expect(component.toJSON()).toMatchSnapshot();
    });

});