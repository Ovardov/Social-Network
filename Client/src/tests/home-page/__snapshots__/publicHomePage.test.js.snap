// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PublicHomePage component should render correctly if user is not logged 1`] = `
<section
  className="container"
>
  <section
    className="content"
  >
    <img
      alt="Friends"
      src="home-page.svg"
    />
    <section
      className="info"
    >
      <h1
        className="name"
      >
        Social Network
      </h1>
      <section
        className="buttons"
      >
        <Link
          className="button"
          to="/register"
        >
          Register
        </Link>
        <Link
          className="button"
          to="/login"
        >
          Login
        </Link>
      </section>
    </section>
  </section>
</section>
`;
